xbmc/addons/ExtsMimeSupportList.cpp

```diff
@@ -28,6 +28,14 @@ CExtsMimeSupportList::CExtsMimeSupportList(CAddonMgr& addonMgr) : m_addonMgr(add
     for (const auto& addonInfo : addonInfos)
       m_supportedList.emplace_back(ScanAddonProperties(ADDON_AUDIODECODER, addonInfo));
   }
+
+  addonInfos.clear();
+
+  if (m_addonMgr.GetAddonInfos(addonInfos, true, ADDON_IMAGEDECODER))
```

Comment:
Maybe it makes sense to overload GetAddonInfos to accept a list of types to return (or a variable number of arguments)? Seems unnecessary to request the list and iterate it twice for each type

Reply:
Would be the best in itself, just not sure whether the overkill of work would be worth it.

A new function could be introduced to support this with vector. Expanding the current one https://github.com/xbmc/xbmc/blob/master/xbmc/addons/AddonManager.cpp#L1160-L1175 would be bad for me, as 99% only need one type and then there is a lot of unnecessary work.

Another way would be to use ADDON_UNKNOWN, but this would transfer all available addons, which would also be more data.

Reply:
Will see to use it as an independent function (without using or changing an existing one)

Reply:
@enen92 have added a new commit where adds a new function on addon manager to get infos by type selection list.